### PR TITLE
Add convenience function to check if a graph is acyclic

### PR DESCRIPTION
**Pull request by borsgeorgica**
Issue https://github.com/igraph/igraph/issues/1932

### CODE REVIEW

File: src/properties/trees.c [hunk 1]
```diff
@@ -601,3 +602,42 @@ igraph_error_t igraph_is_forest(const igraph_t *graph, igraph_bool_t *res,
...
+        // check if it's forest - need to figure out roots
+        IGRAPH_CHECK(igraph_is_forest(graph, &forest, NULL, IGRAPH_ALL));
+        if (forest) {
+            *res = 1;
+            return IGRAPH_SUCCESS;
+        }
+    }
+
+    *res = 0;
+    return IGRAPH_SUCCESS;
```
Comment (szhorvat) [inline src/properties/trees.c:642]: These two lines are not reachable, so they should be removed.

**Comment by szhorvat** [comment]
There is actually a more concise implementation, that makes it very clear that execution can never reach the end of the function:

```
if (igraph_is_directed(graph)) {
    return igraph_is_dag(graph, res);
} else {
    return igraph_is_forest(graph, res, NULL, IGRAPH_ALL);
}
```

**Comment by szhorvat** [comment]
So far it looks good! Could you please add a test, which covers at least the following cases?

 - null graph
 - singleton graph
 - both directed an undirected cases
 - be sure that graphs which contain self-loops are not considered acyclic, in either the directed or undirected case
 - try to think of special cases where things might go wrong, and tests them; self-loops were the example I could think of

**Comment by borsgeorgica** [comment]
> There is actually a more concise implementation, that makes it very clear that execution can never reach the end of the function:
> 
> ```
> if (igraph_is_directed(graph)) {
>     return igraph_is_dag(graph, res);
> } else {
>     return igraph_is_forest(graph, res, NULL, IGRAPH_ALL);
> }
> ```

this is much nicer! thanks

**Comment by szhorvat** [comment]
When you are done, mark this PR as "ready for review". I looked through it and everything looks fine to me.

Others may want to comment on the location where this function is included in the docs.

**Review by szhorvat — APPROVED** [review]

**Comment by ntamas** [comment]
The PR seems okay to me as well; I'll merge it and we can still move the section around in the docs if we find a better place for it.

For instance, there's an entire chapter about "graph cycles" (https://igraph.org/c/html/latest/igraph-Cycles.html), and it barely has any content. Wouldn't it be better to put this function there, along with maybe the entire "Topological sorting, directed acyclic graphs" section and `igraph_girth`?

**Comment by ntamas** [comment]
Thanks for your contribution @borsgeorgica !

**Comment by szhorvat** [comment]
@borsgeorgica Indeed, thanks for the PR!

One of the goals of issue #1932 was as a gentle introduction to igraph internals. Now you have some familiarity with error codes, documentation, and perhaps you also saw how memory management works. If you are interested in continuing to work with igraph, now you are ready for more interesting contributions. You can look through the existing issues, make a proposal for a feature, or simply tell us what you are looking for and we may be able to suggest tasks for which no GitHub issue has been written up yet.

**Comment by szhorvat** [comment]
@ntamas Can you add this to `functions.yaml` before we forget?

**Comment by borsgeorgica** [comment]
Thank you for giving me the opportunity to work on the issue and approving!

I would love to work on other issues as well, it would be nice to learn more about the library, get a better understanding of some of the graph algorithms and get more used to writing C. I was also looking at [this issue](https://github.com/igraph/igraph/issues/1882), would this be a good issue for me to do next? or shall I look at others? Also, I would be more than happy to work on any other problems that don’t have an issue open on Github yet.

**Comment by GroteGnoom** [comment]
#1882 seems fine 🙂 There shouldn't be any surprising difficulties, and you already know how to add a function, so go ahead!

**Pull request merged**